cvmfs/receiver/reactor.cc

```diff
@@ -217,13 +217,13 @@ bool Reactor::HandleGenerateToken(const std::string& req, std::string* reply) {
     return false;
   }
 
-  JsonStringInput input;
+  JsonStringGenerator input;
   input.PushBack("token", session_token.c_str());
   input.PushBack("id", public_token_id.c_str());
   input.PushBack("secret", token_secret.c_str());
+  std::string json = input.GenerateString();
 
-  ToJsonString(input, reply);
-
+  reply = &json;
```

Comment:
If you want to fill the `reply` pointer with data for later use outside the function, I think you need to do `*reply = json`. This way, the `HandleCheckToken` won't modify the `reply` outside of the function. 5x times in the code.

Reply:
I am fixing it now. It was actually something like assign(), I believe...

Reply:
Just `=` would surely be enough, but assign is good as well.

Reply:
Yes, you are right!

I thought I tried it... Oh well...

Reply:
Interesting! https://stackoverflow.com/questions/34196053/stdstringassign-vs-stdstringoperator